src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs

```diff
@@ -377,8 +377,11 @@ internal override void Validate()
         {
             base.Validate();
 
-            //ADFS does not require client id to be in the form of a GUID.
-            if (Config.Authority.AuthorityInfo?.AuthorityType != AuthorityType.Adfs && !Guid.TryParse(Config.ClientId, out _))
+            // ADFS and generic authority do not require client id to be in the form of a GUID.
+            bool isGuidFormatMandatory = 

```

Comment:
There is a unit test that looks at this exception, should be easy to update for generic authority - see https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/main/tests/Microsoft.Identity.Test.Unit/AppConfigTests/PublicClientApplicationBuilderTests.cs#L61

Reply:
You can delete the constants. These aren't really the types of constants people take a dependency on, as this is just a misconfiguration. We will add them back if anybody complains.

Reply:
Completely removed the validation, constants and the related test.